xml/dhcp_vendor_class.xml

```diff
@@ -0,0 +1,215 @@
+    <example hw.family="OfficeJet">Hewlett-Packard OfficeJet</example>
+    <example hw.family="LaserJet">HP LaserJet</example>
+    <example hw.family="Printer">HP Printer</example>
+    <example>Hewlett-Packard JetDirect</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="HP"/>
+    <param pos="1" name="hw.family"/>
+    <param pos="0" name="os.vendor" value="HP"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Mfg=LEXMARK;Typ=(?:MFP|Printer);Mod=Lexmark (\S+);Ser=">
+    <description>Lexmark Printer</description>
+    <example hw.model="MX410de">Mfg=LEXMARK;Typ=MFP;Mod=Lexmark MX410de;Ser=70155PLM0Y5YD;</example>
+    <example hw.model="MS310dn">Mfg=LEXMARK;Typ=Printer;Mod=Lexmark MS310dn;Ser=451432LM0X7XY;</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Lexmark"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="os.vendor" value="Lexmark"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Canon iR-ADV (C?\d+ ?\S*)">
+    <description>Canon imageRunner Printer</description>
+    <example hw.model="C5535 III">Canon iR-ADV C5535 III</example>
+    <example hw.model="C350">Canon iR-ADV C350</example>
+    <example hw.model="4545 III">Canon iR-ADV 4545 III</example>
+    <example hw.model="525">Canon iR-ADV 525</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Canon"/>
+    <param pos="0" name="hw.family" value="imageRunner"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="hw.product" value="{hw.family} {hw.model}"/>
+    <param pos="0" name="os.vendor" value="Canon"/>
+  </fingerprint>
+
+  <fingerprint pattern="Canon (D\d+) Series">
+    <description>Canon imageClas Printer</description>
+    <example hw.model="D1600">Canon D1600 Series</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Canon"/>
+    <param pos="0" name="hw.family" value="imageClass"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="hw.product" value="{hw.family} {hw.model}"/>
+    <param pos="0" name="os.vendor" value="Canon"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Polycom-(VVX\S+)">
+    <description>Polycom IP Phone</description>
+    <example hw.model="VVX410">Polycom-VVX410</example>
+    <param pos="0" name="hw.device" value="VoIP"/>
+    <param pos="0" name="hw.vendor" value="Polycom"/>
+    <param pos="0" name="hw.family" value="VVX"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="os.vendor" value="Polycom"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Cisco Systems, Inc. IP Phone (CP-\d+G?)\.*">
+    <description>Cisco IP Phone</description>
+    <example hw.model="CP-7821">Cisco Systems, Inc. IP Phone CP-7821\u0000</example>
+    <example hw.model="CP-7911G">Cisco Systems, Inc. IP Phone CP-7911G\u0000</example>
```

Comment:
Is that trailing null included in the length of the field or is an artifact of the capturing process?